Sources/Gravatar/Network/Services/AvatarService.swift

```diff
@@ -0,0 +1,49 @@
+import Foundation
+import UIKit
+
+/// AvatarService helps with uploading and downloading avatars.
+public struct AvatarService {
+    private let imageDownloader: ImageDownloader
+    private let imageUploader: ImageUploader
+
+    /// Creates a new `AvatarService`
+    ///
+    /// Optionally, you can pass a custom type conforming to ``HTTPClient`` to gain control over networking tasks.
+    /// Similarly, you can pass a custom type conforming to ``ImageCaching`` to use your custom caching system.
+    /// - Parameters:
+    ///   - client: A type which will perform basic networking operations.
+    ///   - cache: A type which will perform image caching operations.
+    public init(client: HTTPClient? = nil, cache: ImageCaching? = nil) {
+        self.imageDownloader = ImageDownloadService(client: client, cache: cache)
+        self.imageUploader = ImageUploadService(client: client)
+    }
+
+    /// Fetches a Gravatar user profile image using the user account's email, and delivers the image asynchronously. See also: `ImageDownloadService` to
```

Comment:
```suggestion
    /// Fetches a Gravatar user profile image using the user account's email, and delivers the image asynchronously. See also: ``ImageDownloadService`` to
```

With this double quotation marks, the swift-docc will automatically generate a link to the docs page of this symbol :)

Reply:
Aha :TIL: